src/detect-tls-ja3-hash.c

```diff
@@ -56,6 +56,14 @@
 #include "util-unittest.h"
 #include "util-unittest-helper.h"
 
+#ifndef HAVE_JA3
+static int DetectJA3SetupNoSupport(DetectEngineCtx *a, Signature *b, const char *c)
+{
+    SCLogError("no JA3 support built in");
```

Comment:
Wonder if this could be only one function instead of copy pasting 4 lines...

Reply:
TBH I would consider having a generic function in some common file less understandable than this way. By keeping such small static functions in the same file, they are closer to where they are used (in plain developer sight), which I consider much more straightforward than having to dig them up in the IDE. FWIW, if I could even use anonymous functions in C for that, I would ;)

I think code duplication is only bad if it is nontrivial code (with a big impact if it's changed) that is being copied, which is not the case here unless the signature if the `Setup` function pointer would be to change -- but that would touch a lot of places anyway!